vanessaspear/passport-api
# Created a trip photo view

## Description 
The client needs CRUD functionality for adding photos to a trip.  The auth view also needed to be updated to use the email as the username for the login and registration process.

## Changes

- Created a trip photo view 
- Updated auth to use email as username
- Updated auth to add more data to the local storage item

## Requests / Responses

**Request**

POST `/tripphotos` Creates a new trip photo

```json
{
  "trip": 2
  "image": "/media/downtown_nasvhille.jpg"
}
```

**Response**

HTTP/1.1 201 OK

```json
{
  "id": 1,
  "trip": 2
  "image": "/media/downtown_nasvhille.jpg"
}
```
**Request**

PUT `/tripphotos/1` Updates a trip photo

**Response**

HTTP/1.1 204 No content

**Request**

GET `/tripphotos` Gets all trip photos

**Response**

HTTP/1.1 200 OK

```json
[
    {
        "id": 1,
        "trip": {
            "id": 1,
            "name": "Spring Break Trip",
            "city": "Key West",
            "state_or_country": "Florida",
            "departure_date": "2018-03-16",
            "return_date": "2018-03-18",
            "user": 1,
            "reasons": [
                1
            ]
        },
        "image": "/media/key_west_florida.jpg"
    }
]
```

**Request**

GET `/tripphotos/1` Gets a single trip photo

**Response**

HTTP/1.1 200 OK

```json
[
    {
        "id": 1,
        "trip": {
            "id": 1,
            "name": "Spring Break Trip",
            "city": "Key West",
            "state_or_country": "Florida",
            "departure_date": "2018-03-16",
            "return_date": "2018-03-18",
            "user": 1,
            "reasons": [
                1
            ]
        },
        "image": "/media/key_west_florida.jpg"
    }
]
```
**Request**

DELETE `/tripphotos/1` Deletes a trip photo

**Response**

HTTP/1.1 204 No content

## Testing

```
git fetch origin vs-trip-photo-view
git checkout vs-trip-photo-view
```
- Send a GET request to /tripphotos.  Verify response per details above.
- Send a GET request to /tripphotos/1.  Verify response per details above.
- Send a POST request to /tripphotos.  Verify response per details above.
- Send a PUT request to /tripphotos/1 and modify the body.  Verify response per details above.
- Send a DELETE request to /tripphotos/1.  Verify response per details above.

## Related Issues
- Closes #8
